src/main/java/com/booleanuk/Poker.java

```diff
@@ -1,25 +1,161 @@
+        }
+        int maxAt = 0;
+        for (int i = 0; i < handArray.length; i++) {
+            if(handArray[maxAt] < handArray[i]) {
+                maxAt = i;
+            }
+        }
+        if(maxAt != 0) {
+            return hands[maxAt];
+        } else {
+            return new String[]{};
+        }
     }
 
     // Extension 2
     public String[] winningThreeCardHand(String[][] hands) {
-        // Implement the winningThreeCardHand logic here and return the array containing the winning hand to make the tests pass.
-        // You can replace the following return value with something appropriate
-        return new String[]{"Replace me", "with something else"};
+        int[] handArray = new int[hands.length];
+        for(int i = 0; i < hands.length; i++) {
+            handArray[i] = getHighestThreeFromHand(hands[i]);
+        }
+        int maxAt = 0;
+        for (int i = 0; i < handArray.length; i++) {
+            if(handArray[maxAt] < handArray[i]) {
+                maxAt = i;
+            }
+        }
+        if(maxAt != 0) {
+            return hands[maxAt];
+        } else {
+            String[] highestPair = new String[] {"0","0"};
+            for(String[] hand : hands) {
+                if(getHighestPairThreeFromHand(hand) > cards.get((highestPair[0]))) {
+                    highestPair = hand;
+                }
+            }
+            if(highestPair[0].equals("0")) {
+                return new String[]{};
+            } else {
+                return highestPair;
+            }
+
+        }
+    }
+
+    public int getHighestThreeFromHand(String[] hand) {
+        HashMap<String, Integer> cardCountMap = new HashMap<>();
+        for(String card : hand) {
+            cardCountMap.put(card, cardCountMap.getOrDefault(card, 0) + 1);
+        }
+        int highestPair = -1;
+        for(Map.Entry<String, Integer> entry : cardCountMap.entrySet()) {
+            String card = entry.getKey();
+            int value = entry.getValue();
+
+            if(value == 3) {
+                int cardValue = cards.get(card);
+
+                if (cardValue > highestPair) {
+                    highestPair = cardValue;
+                }
+            }
+        }
+        return highestPair;
+    }
+
+    public int getHighestPairThreeFromHand(String[] hand) {
+        HashMap<String, Integer> cardCountMap = new HashMap<>();
+        for(String card : hand) {
+            cardCountMap.put(card, cardCountMap.getOrDefault(card, 0) + 1);
+        }
+        int highestPair = -1;
+        for(Map.Entry<String, Integer> entry : cardCountMap.entrySet()) {
+            String card = entry.getKey();
+            int value = entry.getValue();
+
+            if(value == 2) {
+                int cardValue = cards.get(card);
+
+                if (cardValue > highestPair) {
+                    highestPair = cardValue;
+                }
+            }
+        }
+        return highestPair;
     }
```

Comment:
Your implementation of `winningPair` and the use of a hashmap in `getHighestPairFromHand` is a solid approach. For further improvement, you might consider simplifying the `winningThreeCardHand` method to enhance readability and efficiency. Looking good!